### PR TITLE
fix: data classification in HTML response does not match form settings

### PR DESCRIPTION
**Pull request by craigzour**
# Summary | Résumé

closes https://github.com/cds-snc/platform-forms-client/issues/1934

- Fixed issue where data classification was not properly set in the HTML response. It was static but now relies on the `SecurityAttribute` retrieved from the database.

### CODE REVIEW

File: components/myforms/HTMLDownload/index.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ interface HTMLDownloadProps {
   // submissionID: string;
   responseID: string;
   createdAt: number;
+  securityAttribute: string;
```
Comment (timarney) [inline components/myforms/HTMLDownload/index.tsx:17]: We have a type for this now 

form-types

```
export type SecurityAttribute = "Unclassified" | "Protected A" | "Protected B";
```

**Review by bryan-robitaille — APPROVED** [review]

File: components/myforms/HTMLDownload/ProtectedWarning.tsx [hunk 2]
```diff
@@ -3,20 +3,35 @@ import { useTranslation } from "next-i18next";
...
+  securityAttribute: string;
   lang?: string;
 }
 
 export const ProtectedWarning = (props: ProtectedWarningProps) => {
   const { t, i18n } = useTranslation(["my-forms"]);
   const { lang = i18n.language } = props;
+
+  const renderSecurityAttribute = () => {
+    switch (props.securityAttribute) {
```
Comment (timarney) [inline components/myforms/HTMLDownload/ProtectedWarning.tsx:15]: Something for now or maybe a future PR

In SetResponseDelivery.tsx we have:

```
const classificationOptions = [
  { value: "Unclassified", en: "Unclassified", fr: "Non classifié" },
  { value: "Protected A", en: "Protected A", fr: "Protégé A" },
] as const;
```

So we should figure out how we want to implement so we don't duplicate / have one place to update as need be.

File: pages/api/id/[form]/[submission]/download.tsx [hunk 3]
```diff
@@ -61,23 +61,33 @@ const handler = async (req: NextApiRequest, res: NextApiResponse, props: Middlew
...
+      securityAttribute,
+    } = ((
       vaultResult
     ): {
       formResponse: Responses;
       createdAt: number;
       confirmReceiptCode: string;
       responseID: string;
       submissionID: string;
+      securityAttribute: string;
```
Comment (timarney) [inline pages/api/id/[form]/[submission]/download.tsx:90]: https://github.com/cds-snc/platform-forms-client/pull/1946/files#r1167144926

File: public/static/locales/en/my-forms.json [hunk 4]
```diff
@@ -58,9 +58,11 @@
     "submissionID": "Submission ID",
     "copyResponse": "Copy answer row to clipboard"
   },
-  "protectedBWarning": {
+  "dataClassification": {
     "officialRecord": "This is an official record",
     "retainCopy": "Retain this copy in accordance to your retention policy.",
+    "unclassified": "UNCLASSIFIED",
```
Comment (timarney) [inline public/static/locales/en/my-forms.json:64]: Should these be all uppercase?

We use `Protected A` etc... on other spots

Comment (craigzour) [reply public/static/locales/en/my-forms.json:64]: In the Figma file we can see that the data classification label is written in full uppercase

File: pages/api/id/[form]/[submission]/download.tsx (hunk 3 above)
Comment (craigzour) [reply pages/api/id/[form]/[submission]/download.tsx:90]: Will change

File: components/myforms/HTMLDownload/index.tsx (hunk 1 above)
Comment (craigzour) [reply components/myforms/HTMLDownload/index.tsx:17]: Will change

**Review by thiessenp-cds — COMMENTED** [review]
From testing locally, I can see the default is correctly set to unclassified (in French and English). I'm not sure how to test for protected A/B but the logic looks right to me. I see some other comments/requests here so I'll leave this open.
But from my end LGTM :)

**Comment by timarney** [comment]
@thiessenp-cds 

> From testing locally, I can see the default is correctly set to unclassified (in French and English). I'm not sure how to test for protected A/B but the logic looks right to me. 

1) The default is now Protected A
2) you can update to Protected A or Unclassified via Settings (Protected B isn't available)

Screen Shot 2023-04-14 at 2 36 23 PM

File: components/myforms/HTMLDownload/ProtectedWarning.tsx (hunk 2 above)
Comment (timarney) [reply components/myforms/HTMLDownload/ProtectedWarning.tsx:16]: Ya - we can make another PR to update classificationOptions to use the lang files.

**Review by timarney — APPROVED** [review]

**Pull request merged**